src/main/java/ee/tuleva/onboarding/auth/AuthService.java

```diff
@@ -32,12 +38,37 @@ public String authenticate(GrantType grantType, String authenticationHash) {
       eventPublisher.publishEvent(
           new BeforeTokenGrantedEvent(this, authenticatedPerson, grantType));
 
-      String jwtToken = jwtTokenUtil.generateToken(authenticatedPerson, authorities);
+      String jwtToken = jwtTokenUtil.generateAccessToken(authenticatedPerson, authorities);
+      String refreshToken = jwtTokenUtil.generateRefreshToken(authenticatedPerson, authorities);
 
-      eventPublisher.publishEvent(new AfterTokenGrantedEvent(this, authenticatedPerson, jwtToken));
+      final var tokens = new AccessAndRefreshToken(jwtToken, refreshToken);
+      eventPublisher.publishEvent(new AfterTokenGrantedEvent(this, authenticatedPerson, tokens));
 
-      return jwtToken;
+      return tokens;
     }
     return null;
   }
+
+  public AccessAndRefreshToken refreshToken(String refreshToken) {
+
+    try {
+      TokenType tokenType = jwtTokenUtil.getTypeFromToken(refreshToken);
+      if (tokenType != TokenType.REFRESH) {
+        throw new BadRequestException("Only refresh token is allowed for refresh request.");
```

Comment:
Why are we using `javax.ws.rs.BadRequestException`? Let's use standard Java API `IllegalArgumentException`